test/inductor/test_compiled_autograd.py

```diff
@@ -2429,6 +2429,44 @@ def fn():
 
         self.assertEqual(sum(1 for e in unexpected_logs if e in logs.getvalue()), 0)
 
+    @unittest.expectedFailure
+    def test_saved_tensor_unpack_hook_ordering(self):
+        # not the correct behaviour, I'm just preventing this from changing silently
+        def f(x, y):
+            return x * y
+
+        pack_count = 0
+        unpack_count = 0
+
+        def pack_hook(x):
+            nonlocal pack_count
+            pack_count += 1
+            return x
+
+        def unpack_hook(x):
+            nonlocal unpack_count
+            unpack_count += 1
+            return x
+
+        def tensor_hook(_):
+            # in eager, tensor_hook is fired before unpack_hook
+            # but in compiled autograd, tensor_hook is lifted whereas unpack_hook is not
+            self.assertEqual(unpack_count, 0)
```

Comment:
curious for my own learning: would the current expected failure be the failure of this assert?

Reply:
yes, currently unpack_hook is not lifted, which means it is getting executed during the tracing of the compiled autograd graph. but since tensor_hook is lifted, it is executed after torch.compile traces the graph